src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableAnalyzer.cs

```diff
@@ -0,0 +1,256 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using WTG.Analyzers.Analyzers.BooleanLiteral;
+using WTG.Analyzers.Analyzers.LinqEnumerable;
+using WTG.Analyzers.Utils;
+
+namespace WTG.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class LinqEnumerableAnalyzer : DiagnosticAnalyzer
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+			Rules.DontUseConcatWhenAppendingSingleElementToEnumerablesRule,
+			Rules.DontUseConcatWhenPrependingSingleElementToEnumerablesRule,
+			Rules.DontConcatTwoCollectionsDefinedWithLiteralsRule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction(c => CompilationStart(c));
+		}
+
+		public static bool HasEnumerableAppendMethod(Compilation compilation)
```

Comment:
Why does this class have so many public methods? Are they actually used outside this class? Should they be moved to a separate helper class?